diff --git a/Makefile.am b/Makefile.am
index 49010b3..2c254f8 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -134,6 +134,7 @@ MAN_ROOTS =
 noinst_DATA =
 noinst_HEADERS =
 lib_LTLIBRARIES =
+noinst_LTLIBRARIES =
 noinst_man_MANS =
 noinst_PROGRAMS =
 noinst_SCRIPTS =
diff --git a/lib/automake.mk b/lib/automake.mk
index e4cfc97..b466196 100644
--- a/lib/automake.mk
+++ b/lib/automake.mk
@@ -5,7 +5,10 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without warranty of any kind.
 
-lib_LTLIBRARIES += lib/libopenvswitch.la
+noinst_LTLIBRARIES += lib/libopenvswitch.la
+lib_LTLIBRARIES += lib/libnoiro_openvswitch.la
+lib_libnoiro_openvswitch_la_SOURCES =
+lib_libnoiro_openvswitch_la_LIBADD = lib/libopenvswitch.la
 
 lib_libopenvswitch_la_LIBADD = $(SSL_LIBS)
 lib_libopenvswitch_la_LIBADD += $(CAPNG_LDADD)
@@ -14,7 +17,7 @@ if WIN32
 lib_libopenvswitch_la_LIBADD += ${PTHREAD_LIBS}
 endif
 
-lib_libopenvswitch_la_LDFLAGS = \
+lib_libnoiro_openvswitch_la_LDFLAGS = \
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
         -Wl,--version-script=$(top_builddir)/lib/libopenvswitch.sym \
         $(AM_LDFLAGS)
@@ -326,8 +329,11 @@ nodist_lib_libopenvswitch_la_SOURCES = \
 	lib/vswitch-idl.h
 CLEANFILES += $(nodist_lib_libopenvswitch_la_SOURCES)
 
-lib_LTLIBRARIES += lib/libsflow.la
-lib_libsflow_la_LDFLAGS = \
+noinst_LTLIBRARIES += lib/libsflow.la
+lib_LTLIBRARIES += lib/libnoiro_sflow.la
+lib_libnoiro_sflow_la_SOURCES =
+lib_libnoiro_sflow_la_LIBADD = lib/libsflow.la
+lib_libnoiro_sflow_la_LDFLAGS = \
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
         -Wl,--version-script=$(top_builddir)/lib/libsflow.sym \
         $(AM_LDFLAGS)
diff --git a/lib/libopenvswitch.pc.in b/lib/libopenvswitch.pc.in
index 2a3f2ca..5a0eca4 100644
--- a/lib/libopenvswitch.pc.in
+++ b/lib/libopenvswitch.pc.in
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: libopenvswitch
 Description: Open vSwitch library
 Version: @VERSION@
-Libs: -L${libdir} -lopenvswitch
+Libs: -L${libdir} -lnoiro_openvswitch
 Libs.private: @LIBS@
 Cflags: -I${includedir}/openvswitch
diff --git a/lib/libsflow.pc.in b/lib/libsflow.pc.in
index e70a2b7..6f73164 100644
--- a/lib/libsflow.pc.in
+++ b/lib/libsflow.pc.in
@@ -3,9 +3,9 @@ exec_prefix=@exec_prefix@
 libdir=@libdir@
 includedir=@includedir@
 
-Name: libofproto
+Name: libsflow
 Description: sFlow library of Open vSwitch
 Version: @VERSION@
-Libs: -L${libdir} -lsflow
+Libs: -L${libdir} -lnoiro_sflow
 Libs.private: @LIBS@
 Cflags: -I${includedir}/openvswitch
diff --git a/ofproto/automake.mk b/ofproto/automake.mk
index 7486f2b..1ced5d3 100644
--- a/ofproto/automake.mk
+++ b/ofproto/automake.mk
@@ -5,8 +5,12 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without warranty of any kind.
 
-lib_LTLIBRARIES += ofproto/libofproto.la
-ofproto_libofproto_la_LDFLAGS = \
+noinst_LTLIBRARIES += ofproto/libofproto.la
+lib_LTLIBRARIES += ofproto/libnoiro_ofproto.la
+ofproto_libnoiro_ofproto_la_SOURCES =
+ofproto_libnoiro_ofproto_la_LIBADD = ofproto/libofproto.la
+
+ofproto_libnoiro_ofproto_la_LDFLAGS = \
         -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
         -Wl,--version-script=$(top_builddir)/ofproto/libofproto.sym \
         $(AM_LDFLAGS)
diff --git a/ofproto/libofproto.pc.in b/ofproto/libofproto.pc.in
index 2740712..7a80266 100644
--- a/ofproto/libofproto.pc.in
+++ b/ofproto/libofproto.pc.in
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: libofproto
 Description: OpenFlow library of Open vSwitch
 Version: @VERSION@
-Libs: -L${libdir} -lofproto
+Libs: -L${libdir} -lnoiro_ofproto
 Libs.private: @LIBS@
 Cflags: -I${includedir}/openvswitch
diff --git a/tests/automake.mk b/tests/automake.mk
index d42b576..e1782a6 100644
--- a/tests/automake.mk
+++ b/tests/automake.mk
@@ -370,7 +370,7 @@ tests_ovstest_SOURCES += \
 	tests/test-netlink-conntrack.c
 endif
 
-tests_ovstest_LDADD = lib/libopenvswitch.la ovn/lib/libovn.la
+tests_ovstest_LDADD = lib/libnoiro_openvswitch.la ovn/lib/libovn.la
 dist_check_SCRIPTS = tests/flowgen.pl
 
 noinst_PROGRAMS += tests/test-strtok_r
